client/state/form/reducer.js

```diff
@@ -34,9 +27,13 @@ export default function form( state = {}, action ) {
 		} );
 	}
 
-	// Allow client-side form validation to take over error state when inputs change
-	if ( settingsActions.hasOwnProperty( action.type ) && newState.hasOwnProperty( 'errors' ) ) {
-		delete newState.errors;
+	if ( settingsActions.hasOwnProperty( action.type ) ) {
+		newState.pristine = false;
```

Comment:
It seems like it would be simpler for this to be part of the SET_FIELD reducer instead.

The SET_FIELD reducer could replace this

```
if ( 'success' === action.field && action.value ) {
   newObj.pristine = true;
}```

with something like this then
```
newObj.pristine = ( 'success' === action.field && action.value )
```